tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -4,18 +4,94 @@ import (
 	"fmt"
 	"strconv"
 	"strings"
+	"time"
 
 	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
 	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	nodestat "github.com/rancher/rancher/tests/framework/extensions/nodes"
+	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
 )
 
+const (
+	active = "active"
+)
+
+// MatchRoleToConfig matches the role of machinePools to the nodeRoles.
+func MatchRoleToConfig(nodeRoles NodeRoles, machinePools []apisV1.RKEMachinePool) (int, int32) {
+	count := int32(0)
+	for index, machinePoolConfig := range machinePools {
+		if nodeRoles.ControlPlane != machinePoolConfig.ControlPlaneRole {
+			continue
+		}
+		if nodeRoles.Etcd != machinePoolConfig.EtcdRole {
+			continue
+		}
+		if nodeRoles.Worker != machinePoolConfig.WorkerRole {
+			continue
+		}
+
+		count += *machinePoolConfig.Quantity
```

Comment:
Just to make sure I am understanding this bit; Is this sum working as a sort of fallback with the 0 initial?

Reply:
That is correct. We're using `count` as a counter that is going to append the desired machine pool's quantity. That way, we later can make sure that we are scaling up and down by this returned value.

This is needed because say a user has a cluster of 3 etcd, 2 cp and 3 worker. With the static test cases, the defined quantity will properly scale up and down without complications.